js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -332,14 +337,27 @@ export const useMaterializationAction = (preferredJobName?: string) => {
     }
     setState({type: 'loading'});
 
-    const data = await onLoad(assetKeysOrJob);
+    let data = await onLoad(assetKeysOrJob);
 
     if ('assetNodeDefinitionCollisions' in data && data.assetNodeDefinitionCollisions.length) {
       showCustomAlert(buildAssetCollisionsAlert(data));
       setState({type: 'none'});
       return;
     }
 
+    if ('assetNodeAdditionalRequiredKeys' in data && data.assetNodeAdditionalRequiredKeys.length) {
+      setState({type: 'none'});
```

Comment:
Maybe this setState actually belongs on the catch on line 353, and then we remove the setstate on line 355 and rely on the one on line 338

Reply:
Ahh yeah I'm down to do that, I think it'd mean it wouldn't stop while the confirm is open and that seems better